src/Overlay.tsx

```diff
@@ -65,6 +65,7 @@ export type OverlayProps = {
   onClickOutside: (e: TouchOrMouseEvent) => void
   onEscape: (e: KeyboardEvent) => void
   visibility?: 'visible' | 'hidden'
+  'data-test-id'?: string
```

Comment:
This does feel very specific to the use case we are trying to solve.  I did some research and it looks like template literals would be great here, like ``[key: `data-${string}`]: string``, but that isn't supported at this time (https://github.com/microsoft/TypeScript/pull/26797).  As a workaround, we could allow addition keys with something like `[additionalKey: string]: unknown`

Reply:
Great idea, thanks for weighing in.